pkg/ipamd/rpc_handler.go

```diff
@@ -72,17 +146,44 @@ func (s *server) AddNetwork(ctx context.Context, in *rpc.AddNetworkRequest) (*rp
 		DeviceNumber:    int32(deviceNumber),
 		UseExternalSNAT: useExternalSNAT,
 		VPCcidrs:        pbVPCcidrs,
+		PodVlanId:       int32(vlanId),
+		PodENIMAC:       branchENIMAC,
+		PodENISubnetGW:  podENISubnetGW,
+		ParentIfIndex:   int32(trunkENILinkIndex),
 	}
 
 	log.Infof("Send AddNetworkReply: IPv4Addr %s, DeviceNumber: %d, err: %v", addr, deviceNumber, err)
-	addIPCnt.Inc()
 	return &resp, nil
 }
 
 func (s *server) DelNetwork(ctx context.Context, in *rpc.DelNetworkRequest) (*rpc.DelNetworkReply, error) {
 	log.Infof("Received DelNetwork for Sandbox %s", in.ContainerID)
 	delIPCnt.With(prometheus.Labels{"reason": in.Reason}).Inc()
-
+	if s.ipamContext.enablePodENI {
+		pod, err := s.ipamContext.GetPod(in.K8S_POD_NAME, in.K8S_POD_NAMESPACE)
+		if err != nil {
+			if k8serror.IsNotFound(err) {
+				log.Warn("Send AddNetworkReply: pod not found")
+				return &rpc.DelNetworkReply{Success: true}, nil
```

Comment:
Does CNI takes care if you send empty IP in the response?

Reply:
Yes, if we can't parse the IP returned, we just log a warning and exit. From the `del` function:
```
	deletedPodIP := net.ParseIP(r.IPv4Addr)
	if deletedPodIP != nil {
 .
 .
 .
	} else {
		log.Warnf("Container %s did not have a valid IP %s", args.ContainerID, r.IPv4Addr)
	}
	return nil
}
```